arrow/src/main/java/org/apache/iceberg/arrow/vectorized/NullabilityHolder.java

```diff
@@ -52,10 +52,12 @@ public void setNotNull(int index) {
   }
 
   public void setNulls(int startIndex, int num) {
+    // TODO: do we have to modify numNulls?
```

Comment:
I'd appreciate if someone could explain why we only modify `numNulls` in `setNull`.

Reply:
It definitely seems like we should be adding to numNulls here. The only consumers of this I see are in the classes Extending the Spark ColumnVector class which does expect this value to be correct.

Reply:
@samarthjain can you comment on this?

Reply:
Yeah, looks like we should add to `numNulls` here.

Reply:
I think it is the only place we need to change in this class.

Reply:
The challenge is also what number we should add. Are there overlaps, for example? I don't think we can answer that without iterating through the array.

Reply:
We can make the same assumption as in `setNull` (i.e. the set indices don't overlap) and then simply add the number of set nulls. I'd probably just call `numNulls += num`.

Thoughts, @samarthjain @RussellSpitzer @rdblue?

Reply:
We currently just assume the caller is carful, I think we can just keep that assumption

Reply:
Agreed, we should be adding to `numNulls` here. Looking at the caller of `setNulls()`, it looks like we call this method in cases where RLE encoding and the definition level tells us that next num values are null. So it should be ok to do `numNulls += num`

Reply:
Fixed.